### PR TITLE
fast functor updated

### PR DESCRIPTION
**Pull request by hdgarrood**
This is @matthewleon's faster `List.map`, from #135, updated for 0.12.

### CODE REVIEW

File: src/Data/List/Types.purs [hunk 1]
```diff
@@ -67,7 +73,31 @@ instance monoidList :: Monoid (List a) where
...
+    unrolledMap :: List a -> List b
+    unrolledMap (x1 : x2 : Nil) = f x1 : f x2 : Nil
+    unrolledMap (x1 : Nil) = f x1 : Nil
+    unrolledMap _ = Nil
+
+    reverseUnrolledMap :: List (List a) -> List b -> List b
+    reverseUnrolledMap ((x1 : x2 : x3 : _) : cs) acc =
+      reverseUnrolledMap cs (f x1 : f x2 : f x3 : acc)
+    -- if we pattern match on Nil, we need a Partial constraint,
+    -- which kills TCO
```
Comment (natefaubion) [inline src/Data/List/Types.purs]: FWIW I don't think partial constraints kill tco anymore.

Comment (hdgarrood) [reply src/Data/List/Types.purs]: Good point. Shall we just remove this comment then?

Comment (natefaubion) [reply src/Data/List/Types.purs]: 👍

**Comment by matthewleon** [comment]
Cool to see this happen!

**Comment by hdgarrood** [comment]
@matthewleon thanks again for this, and sorry it slipped under the radar for so long!

**Pull request merged**